src/params.rs

```diff
@@ -61,6 +62,26 @@ pub struct RunParameters {
                             // HOME: /
 }
 
+impl RunParameters {
+    /// Examines the local network interfaces, and tries to find our assigned IP within the data network.
+    /// If running in a sidecar-less environment, the loopback address is returned.
+    pub fn data_network_ip(&self) -> std::io::Result<Option<IpAddr>> {
+        if !self.test_sidecar {
+            // this must be a local:exec runner and we currently don't support traffic shaping on it for now, just return the loopback address.
```

Comment:
```suggestion
            // This must be a local:exec runner and we currently don't support traffic shaping on it for now, just return the loopback address.
```

Mind wrapping here at 80 or 120 chars?